web/src/redux/slices/DAG.slice.ts

```diff
@@ -0,0 +1,68 @@
+import {createSlice, PayloadAction} from '@reduxjs/toolkit';
+import {Node, Edge, NodeChange, applyNodeChanges} from 'react-flow-renderer';
+
+import DAGModel from 'models/DAG.model';
+import {TSpan} from 'types/Span.types';
+import {clearAffectedSpans, setAffectedSpans, setMatchedSpans, setSelectedSpan} from './Span.slice';
+import {setSelectedAssertion} from './TestDefinition.slice';
+
+interface IState {
+  edges: Edge[];
+  nodes: Node[];
+}
+
+const initialState: IState = {
+  edges: [],
+  nodes: [],
+};
+
+const dagSlice = createSlice({
```

Comment:
This is the DAG redux slice that contains `edges` and `nodes`. Here we update the state based on user events (spanSelected, search, affectedSpans, ...)